authentication/active_directory_authentication_provider.py

```diff
@@ -0,0 +1,46 @@
+#########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from ldap_authentication_provider import LDAPAuthenticationProvider
+
+DEFAULT_AUTH_ATTR_NAME = 'sAMAccountName'
+
+
+class ActiveDirectoryAuthenticationProvider(LDAPAuthenticationProvider):
+
+    def __init__(self, ldap_url, domain_name=None, search_properties=None):
+        super(ActiveDirectoryAuthenticationProvider, self)\
+            .__init__(ldap_url, search_properties)
+        self.domain_name = domain_name
+
+    def authenticate(self, userstore=None):
+        who, password = self.get_creds_from_request()
+
+        if self.search_props:
+            self.search_props.setdefault('user_id_attribute',
+                                         DEFAULT_AUTH_ATTR_NAME)
+            user_dn = self.get_dn_by_user_attribute(who,
+                                                    self.ldap_url,
+                                                    **self.search_props)
+        else:
+            self._logger.debug('Attempting to connect using the user id '
```

Comment:
1. if self.domain_name is None you might get an exception here..?
2. I don't see why we only log when search is not used... I think it's better to move the logging to line 44, just before calling validate_bind, and change the message to:
self._logger.debug('Attempting to connect using: {user_dn}'.format(user_dn))
because it includes the domain anyways. also, I removed the 'the user id' part from the message because it might be a UPN or something else.